internal/generate/run_test.go

```diff
@@ -110,308 +110,145 @@ func TestRun(t *testing.T) {
-		options := generate.CLIOptions{
-			Destdir:      destdir,
-			File:         filepath.Join(testdata, "success_with_options", "types.go"),
-			Structs:      []string{"Options", "Empty"},
-			ValidateFunc: "Validate",
-		}
-
-		// Act
-		err := generate.Run(pwd, options)
-
-		// Assert
-		assert.NoError(t, err)
-		testfs.AssertEqualDir(t, assertdir, destdir)
-	})
-
-	t.Run("success_return_copy", func(t *testing.T) {
-		// Arrange
-		assertdir := filepath.Join(testdata, "success_return_copy", "builders")
-		destdir := filepath.Join(t.TempDir(), "builders")
-		options := generate.CLIOptions{
-			Destdir:    destdir,
-			File:       filepath.Join(testdata, "success_return_copy", "types.go"),
-			Structs:    []string{"ReturnCopy"},
-			ReturnCopy: true,
-		}
+	for _, tc := range []struct {
+		generate.CLIOptions
+		DirName     string
+		SamePackage bool
+	}{
+		{
+			DirName: "success_channels",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"Chan"},
+			},
+		},
+		{
+			DirName: "success_export",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"Export"},
+			},
+		},
+		{
+			DirName: "success_funcs",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"Func"},
+			},
+		},
+		{
+			DirName: "success_interface",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"Interface", "InterfaceNoFields"},
+			},
+		},
+		{
+			DirName: "success_maps",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"Map"},
+			},
+		},
+		{
+			DirName: "success_module_replace",
+			CLIOptions: generate.CLIOptions{
+				File:    "module::github.com/sirupsen/logrus/hooks/test/test.go",
+				Structs: []string{"Hook"},
+			},
+		},
+		{
+			DirName: "success_module_root",
+			CLIOptions: generate.CLIOptions{
+				File:    "module::github.com/go-playground/validator/v10/errors.go",
+				Structs: []string{"InvalidValidationError"},
+			},
+		},
+		{
+			DirName: "success_module_subdir",
+			CLIOptions: generate.CLIOptions{
+				File:    "module::github.com/sirupsen/logrus/hooks/test/test.go",
+				Structs: []string{"Hook"},
+			},
+		},
+		{
+			DirName: "success_naming",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"Naming"},
+			},
+		},
+		{
+			DirName: "success_root_gomod",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"RootType"},
+			},
+		},
+		{
+			DirName: "success_same_package",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"SamePackage", "unexportedType"},
+			},
+			SamePackage: true,
+		},
+		{
+			DirName: "success_same_package_prefix",
+			CLIOptions: generate.CLIOptions{
+				Structs:      []string{"unexportedTypePrefix"},
+				SetterPrefix: "Set",
+			},
+			SamePackage: true,
+		},
+		{
+			DirName: "success_slices",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"ArrayAndSlice"},
+			},
+		},
+		{
+			DirName: "success_struct",
+			CLIOptions: generate.CLIOptions{
+				Structs: []string{"Struct", "StructNoFields"}},
```

Comment:
```suggestion
				Structs: []string{"Struct", "StructNoFields"}
				},
```
**todo :** Linting issue